src/game/DBScripts/ScriptMgr.cpp

```diff
@@ -2616,8 +2703,164 @@ bool ScriptAction::ExecuteDbscriptCommand(WorldObject* pSource, WorldObject* pTa
             creature->AI()->AttackClosestEnemy();
             break;
         }
+        case SCRIPT_COMMAND_SPAWN_GROUP:                    // 60
+        {
+            if (LogIfNotCreature(pTarget))
+                return false;
+
+            Creature* leader = static_cast<Creature*>(pTarget);
+
+            auto groupData = leader->GetCreatureGroup();
+            auto currSlot = leader->GetFormationSlot();
+            FormationData* formationData = nullptr;
+            if (currSlot)
+                formationData = currSlot->GetFormationData();
+
+            switch (m_script->formationData.command)
+            {
+            case 2:                         // set formation
+            {
+                if (LogIfNotCreature(pSource))
+                    return false;
+
+                CreatureGroup* grpData = nullptr;
+                if (!m_script->formationData.data1)
+                {
+                    if (!groupData)
+                    {
+                        sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` script id %u, command %u and subcommand formation create(2) failed. Target group(%u) not found!",
+                            m_table, m_script->id, m_script->command, m_script->formationData.data1);
+                        break;
+                    }
+                    groupData->SetFormationData(nullptr);
+                    break;
+                }
+                else
+                {
+                    auto sgData = leader->GetMap()->GetSpawnManager().GetSpawnGroup(m_script->formationData.data1);
+                    if (!sgData || !sgData->GetCreatureGroup())
+                    {
+                        sLog.outErrorDb(" DB-SCRIPTS: Process table `%s` script id %u, command %u and subcommand formation create(2) failed. Target group(%u) not found!",
+                            m_table, m_script->id, m_script->command, m_script->formationData.data1);
+                        break;
+                    }
+                    grpData = sgData->GetCreatureGroup();
+                }
+
+                if (grpData->GetFormationData())
+                {
+                    //
+                    break;
+                }
+
+                FormationEntrySPtr fEntry = std::make_shared<FormationEntry>();
```

Comment:
Why does this need to be shared ptr at entry level and not at global level. Not inherently wrong just curious.

Reply:
Hum yes maybe not needed too here